Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java

```diff
@@ -339,20 +357,28 @@ private FileSec generateFileSec(Map<URI, FileType> mediaFilesToIDFiles) {
      *            In this map, the function returns the assigned identifier for
      *            each media unit so that the link pairs of the struct link
      *            section can be formed later.
+     * @param mets
+     *            the METS structure in which the meta-data is added
      * @return the physical struct map
      */
     private StructMapType generatePhysicalStructMap(
-            Map<URI, FileType> mediaFilesToIDFiles, Map<MediaUnit, String> mediaUnitIDs) {
+            Map<URI, FileType> mediaFilesToIDFiles, Map<MediaUnit, String> mediaUnitIDs, MetsType mets) {
         StructMapType physical = new StructMapType();
         physical.setTYPE("PHYSICAL");
-        DivType boundBook = new DivType();
-        for (MediaUnit mediaUnit : workpiece.getMediaUnits()) {
-            boundBook.getDiv().add(new FileXmlElementAccess(mediaUnit).toDiv(mediaFilesToIDFiles, mediaUnitIDs));
-        }
-        physical.setDiv(boundBook);
+        physical.setDiv(
+            generatePhysicalStructMapRecursive(workpiece.getMediaUnit(), mediaFilesToIDFiles, mediaUnitIDs, mets));
         return physical;
     }
 
+    private DivType generatePhysicalStructMapRecursive(MediaUnit mediaUnit, Map<URI, FileType> mediaFilesToIDFiles,
+            Map<MediaUnit, String> mediaUnitIDs, MetsType mets) {
```

Comment:
The 
generateFileSecRecursive is actually recursive, but where is the recursive part here?